### PR TITLE
[xdl] Handle errors when trying to sign the manifest

### PR DESCRIPTION
**Pull request by fson**
When the `ManifestHandler` tries to sign the manifest (requested by the client), automatically handle typical errors and fall back to offline mode:
- The project has an `owner` specified, but you lack permissions to that account -> fall back to offline mode! Fixes https://github.com/expo/expo-cli/issues/3062.
  Screen Shot 2021-01-28 at 17 02 17

- You don't have an internet connection, which results in `Error: getaddrinfo ENOTFOUND exp.host` -> fall back to offline mode! Fixes https://github.com/expo/expo-cli/issues/238.
  Screen Shot 2021-01-28 at 16 59 05

### CODE REVIEW

File: packages/xdl/src/project/ManifestHandler.ts [hunk 1]
```diff
@@ -240,6 +266,21 @@ export async function getManifestResponseAsync({
...
 
+const addSigningDisabledWarning = (() => {
+  let seen = false;
+  return (projectRoot: string, reason: string) => {
+    if (!seen) {
+      seen = true;
+      ProjectUtils.logWarning(
+        projectRoot,
+        'expo',
+        reason + '\n' + `Falling back to offline mode.`,
```
Comment (brentvatne) [inline packages/xdl/src/project/ManifestHandler.ts:277]:
```suggestion
        reason + '\nFalling back to offline mode.`,
```

File: packages/xdl/src/project/ManifestHandler.ts [hunk 2]
```diff
@@ -231,7 +233,31 @@ export async function getManifestResponseAsync({
...
+    manifestString = await getManifestStringAsync(manifest, hostInfo.host, acceptSignature);
+  } catch (error) {
+    if (error.code === 'UNAUTHORIZED_ERROR' && manifest.owner) {
+      // Don't have permissions for siging, warn and enable offline mode.
+      addSigningDisabledWarning(
+        projectRoot,
+        `This project belongs to ${chalk.bold(
+          `@${manifest.owner}`
+        )} and you have not been granted the appropriate permissions.\n` +
+          `Please request access from an admin of @${manifest.owner} or change the \"owner\" field to an account you belong to.\n` +
```
Comment (brentvatne) [inline packages/xdl/src/project/ManifestHandler.ts:247]: github warnings here are accurate re: escape character

**Review by brentvatne — APPROVED** [review]
looks good, i think this is a nice solution to our owner field problems. just need to fix the warnings raised by linter then ship it

File: packages/xdl/src/project/ManifestHandler.ts [hunk 3]
```diff
@@ -231,7 +233,31 @@ export async function getManifestResponseAsync({
...
+          learnMore('https://docs.expo.io/versions/latest/config/app/#owner')
+      );
+      Config.offline = true;
+      manifestString = await getManifestStringAsync(manifest, hostInfo.host, acceptSignature);
+    } else if (error.code === 'ENOTFOUND') {
+      // Got a DNS error, i.e. can't access exp.host, warn and enable offline mode.
+      addSigningDisabledWarning(projectRoot, 'Not connected to internet.');
```
Comment (byCedric) [inline packages/xdl/src/project/ManifestHandler.ts:254]: For users in countries where you might need a VPN, this could be confusing. They could have access to "the internet" but not `exp.host`. 

- Maybe rewording this to "Could not reach the Expo servers" might be better?
- Or even better, add tips for the user to manually test/fix connection (e.g. "Could not reach Expo servers, please check if you can access exp.host")

Comment (fson) [reply packages/xdl/src/project/ManifestHandler.ts:254]: Great point, I'll update the wording here.
